### PR TITLE
DM-14391 Simplify initialization of display_firefly and firefly_client

### PR DESCRIPTION
**Pull request by stargaser**
Changes to the `display_firefly` backend to enable easier initialization. These changes go together with the 2.0.0+ version of `firefly_client`.

* set default `name` to None, to allow `firefly_client` to auto-initialize the channel.
* change `host` to `url` to track the new initialization of `firefly_client.FireflyClient`.
* leave default `url` unspecified, to allow `FireflyClient` to initialize from environment variables, as will be provided in LSST Science Platform environments.
* upon display creation, attempt to launch a browser, which now displays instructions for the user if the browser open is unsuccessful.
* expose the `FireflyClient` instance and the browser url as attributes.
* update the user instructions in the documentation.
* reorganize the documentation to make it ready for publication on pipelines.lsst.io.

### CODE REVIEW

File: doc/display_firefly/index.rst [hunk 1]
```diff
@@ -0,0 +1,273 @@
+.. _display_firefly:
```
Comment (SimonKrughoff) [inline doc/display_firefly/index.rst]: I think doing a `git mv` would have let you preserve the diff.

File: python/lsst/display/firefly/firefly.py [hunk 2]
```diff
@@ -89,13 +88,20 @@ def __init__(self, display, verbose=False, host="http://localhost:8080",
         global _fireflyClient
         if not _fireflyClient:
             try:
-                _fireflyClient = firefly_client.FireflyClient(host,
-                                        channel=name, basedir=basedir, **kwargs)
+                if url is None:
+                    _fireflyClient = firefly_client.FireflyClient(channel=name, **kwargs)
+                else:
+                    _fireflyClient = firefly_client.FireflyClient(url,
+                                        channel=name, **kwargs)
```
Comment (SimonKrughoff) [inline python/lsst/display/firefly/firefly.py]: According to our standards, I think `channel` should be aligned with `url`.

File: python/lsst/display/firefly/firefly.py [hunk 3]
```diff
@@ -89,13 +88,20 @@ def __init__(self, display, verbose=False, host="http://localhost:8080",
...
             except Exception as e:
```
Comment (SimonKrughoff) [inline python/lsst/display/firefly/firefly.py]: Typically it's better to trap the exceptions you expect.  Trapping a bare exception means you can't tell what actually went wrong when something unexpected happens.

Comment (stargaser) [reply python/lsst/display/firefly/firefly.py]: Agreed. I'll trap on the two exceptions that are raised, one for a nonsense url and one for a working web server that is not a Firefly url.

**Pull request merged**